src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -171,6 +171,7 @@ protected virtual TransactionResult Execute(Transaction tx, in BlockExecutionCon
             long gasAvailable = tx.GasLimit - intrinsicGas.Standard;
             ExecuteEvmCall(tx, header, spec, tracer, opts, delegationRefunds, intrinsicGas.FloorGas, accessTracker, gasAvailable, env, out TransactionSubstate? substate, out GasConsumed spentGas, out byte statusCode);
             PayFees(tx, header, spec, tracer, substate, spentGas.SpentGas, premiumPerGas, blobBaseFee, statusCode);
+            tx.SpentGas = spentGas.SpentGas;
```

Comment:
How `SpentGas` was not needed before ?

Reply:
Is using the prior build build to inform the next block build; isn't needed outside block production